clients/python-wrapper/lakefs/object.py

```diff
@@ -399,9 +400,6 @@ def __init__(self,
                  metadata: Optional[dict[str, str]] = None,
                  client: Optional[Client] = None) -> None:
 
-        if 'x' in mode and obj.exists():  # Requires explicit create
-            raise ObjectExistsException
-
```

Comment:
Discussed with @nopcoder and decided to reintroduce this check to fail early.

While this may impact performance, this check already existed, so this PR is not introducing any additional overhead.